src/main/java/com/dubture/jenkins/digitalocean/ComputerLauncher.java

```diff
@@ -189,29 +189,45 @@ private int bootstrap(Connection bootstrapConn, Computer computer, PrintStream l
             }
             if (!isAuthenticated) {
                 logger.println("Authentication failed");
-                return FAILED;
+                return BootstrapResult.FAILED;
             }
             closeBootstrap = false;
-            return SAMEUSER;
+            return BootstrapResult.SAMEUSER;
         } catch (Exception e) {
             e.printStackTrace(logger);
-            return FAILED;
+            return BootstrapResult.FAILED;
         } finally {
             if (closeBootstrap)
                 bootstrapConn.close();
         }
     }
 
-    private Connection connectToSsh(Computer computer, PrintStream logger) throws InterruptedException, RequestUnsuccessfulException, DigitalOceanException {
+    private Connection connectToSsh(Computer computer, PrintStream logger) throws RequestUnsuccessfulException, DigitalOceanException {
 
         // TODO: make configurable?
         final long timeout = TimeUnit2.MINUTES.toMillis(5);
         final long startTime = System.currentTimeMillis();
+        final int sleepTime = 10;
 
-        while(true) {
+        while (true) {
             try {
+                Droplet droplet = DigitalOcean.getDroplet(computer.getCloud().getAuthToken(), computer.getNode().getDropletId());
+
+                switch (droplet.getStatus()) {
+                    case ACTIVE:
+                        break;
+
+                    case ARCHIVE:
+                    case OFF:
+                        throw new IllegalStateException("Droplet has unexpected status: " + droplet.getStatus());
+
+                    case NEW:
+                        logger.println("Droplet in NEW state, waiting for it to become ACTIVE");
+                        throw new IOException("sleep");
```

Comment:
tbh I dont like the approach of handling a regular Situation by an exception. sleep cont. should work fine here too, wouldn't it?